vpr/src/base/vpr_constraints_serializer.h

```diff
@@ -65,6 +70,11 @@ struct VprConstraintsContextTypes : public uxsd::DefaultVprConstraintsContextTyp
 
 class VprConstraintsSerializer final : public uxsd::VprConstraintsBase<VprConstraintsContextTypes> {
   public:
+    //VprConstraintsSerializer() : report_error_(nullptr) {}
+    VprConstraintsSerializer(VprConstraints constraints)
+        : constraints_(constraints)
```

Comment:
pass in by reference instead of copying in (the constraints object)

Reply:
when you call the writer, call with a reference to the constraints object so that it doesn't copy the whole thing. Can also make it a pointer.

Reply:
Don't put it into the constructor if you don't have to (because during loading you don't need to pass in a constraints object). Instead, pass it by reference to the top level writing routine.

Reply:
disregard above, discussed that constraints object is need for both load and write. But have two constructors so you don't have to pass in an empty object when doing load_vpr_constraints